deployment/helpers.go

```diff
@@ -108,3 +110,45 @@ func ParseErrorFromABI(errorString string, contractABI string) (string, error) {
 	}
 	return "", errors.New("error not found in ABI")
 }
+
+// ContractDeploy represents the result of an EVM contract deployment
+// via an abigen Go binding. It contains all the return values
+// as they are useful in different ways.
+type ContractDeploy[C any] struct {
+	Address  common.Address     // We leave this incase a Go binding doesn't have Address()
+	Contract C                  // Expected to be a Go binding
+	Tx       *types.Transaction // Incase the caller needs for example tx hash info for
+	Tv       TypeAndVersion
+	Err      error
+}
+
+// DeployContract deploys an EVM contract and
+// records the address in the provided address book
+// if the deployment was confirmed onchain.
+// Deploying and saving the address is a very common pattern
+// so this helps to reduce boilerplate.
+// It returns an error if the deployment failed, the tx was not
+// confirmed or the address could not be saved.
+func DeployContract[C any](
```

Comment:
Intending to re-use this helper in deployment/common for MCMS, capability registry, link/weth token and any other product agnostic contracts